app/services/candidates/registrations/subject_and_date_information.rb

```diff
@@ -8,6 +8,8 @@ class SubjectAndDateInformation < RegistrationStep
       attribute :bookings_subject_id, :integer
 
       validates :bookings_subject_id, presence: true, if: :for_subject_specific_date?
+      validates :placement_date_subject, presence: true, if:
+        -> { for_subject_specific_date? && bookings_subject_id }
```

Comment:
I don't think we need the && condition. `for_subject_specific_date?` checks the date is subject_specific and subject_specific dates have a validation that subjects are present